variables.tf

```diff
@@ -1,3 +1,15 @@
+variable "allow_publish_aws_services" {
```

Comment:
`allow_publish_aws_services` and `allow_publish_iam_arns` should be type `bool`.

Add 2 more variables: `allowed_aws_services` and `allowed_iam_arns` as `list(string)`.

The main reason for this change is that such values may come from computed sources (eg module's output) and Terraform will fail when they are used in `count` and `for_each`.

Reply:
Please, rearrange description, type, default to being consistent in most other places.

Reply:
Oof, you're right I forgot about that 🤦‍♂️   Although I'm not a big fan of the bool values that "enable" list parameters.  By changing around the logic of `policy` vs `allow_publish_*` I was able to avoid the computed values limitation without introducing any extra bools.  Let me know what you think!